openmetadata-ui/src/main/resources/ui/src/pages/ClassificationVersionPage/ClassificationVersionPage.test.tsx

```diff
@@ -0,0 +1,184 @@
+  default: jest
+    .fn()
+    .mockImplementation(({ children }) => <div>{children}</div>),
+}));
+
+jest.mock('../../components/PermissionProvider/PermissionProvider', () => ({
+  usePermissionProvider: jest.fn().mockImplementation(() => ({
+    getEntityPermissionByFqn: jest
+      .fn()
+      .mockImplementation(() => Promise.resolve(ENTITY_PERMISSIONS)),
+  })),
+}));
+
+jest.mock('../../utils/PermissionsUtils', () => ({
+  checkPermission: jest.fn().mockReturnValue(true),
+  DEFAULT_ENTITY_PERMISSION: {
+    ViewAll: true,
+    ViewBasic: true,
+  },
+}));
+
+jest.mock('../../rest/tagAPI', () => ({
+  getClassificationByName: jest
+    .fn()
+    .mockImplementation(() => Promise.resolve(MOCK_ALL_CLASSIFICATIONS)),
+  getClassificationVersionData: jest
+    .fn()
+    .mockImplementation(() => Promise.resolve(MOCK_ALL_CLASSIFICATIONS)),
+  getClassificationVersionsList: jest
+    .fn()
+    .mockImplementation(() => Promise.resolve(MOCK_ALL_CLASSIFICATIONS)),
+}));
+
+jest.mock('../../utils/RouterUtils', () => ({
+  getClassificationVersionsPath: jest.fn(),
+  getClassificationDetailsPath: jest.fn(),
+}));
+
+jest.mock('../../utils/ToastUtils', () => ({
+  showErrorToast: jest.fn(),
+}));
+
+describe('ClassificationVersionPage component', () => {
+  it('should render Loader', async () => {
+    render(<ClassificationVersionPage />, {
+      wrapper: MemoryRouter,
+    });
+
+    const loader = screen.getByText('Loader component');
+
+    expect(loader).toBeInTheDocument();
+  });
+
+  it('should render EntityVersionTimeLine', async () => {
+    await act(async () => {
+      render(<ClassificationVersionPage />, {
+        wrapper: MemoryRouter,
+      });
+    });
+
+    const versionData = await screen.findByTestId('version-data');
+    const entityVersionTimeLine = await screen.findByText(
+      'testEntityVersionTimeLine'
+    );
+
+    expect(versionData).toBeInTheDocument();
+    expect(entityVersionTimeLine).toBeInTheDocument();
+  });
+
+  it('should renders ErrorPlaceHolder with permission error', async () => {
+    mockedUsePermissionProvider.mockReturnValue({
+      getEntityPermissionByFqn: jest.fn().mockResolvedValue({}),
+    });
+
+    render(<ErrorPlaceHolder type={ERROR_PLACEHOLDER_TYPE.PERMISSION} />, {
+      wrapper: MemoryRouter,
+    });
+
+    const errorPlaceholder = await screen.findByText('testErrorPlaceHolder');
+
+    expect(errorPlaceholder).toBeInTheDocument();
+  });
+
+  it('should renders ClassificationDetails in version view with all permissions', () => {
+    const currentVersionData = {
+      description: 'description',
+      name: 'advanceSearch',
+    };
+
+    render(
+      <ClassificationDetails
+        isVersionView
+        classificationPermissions={DEFAULT_ENTITY_PERMISSION}
+        currentClassification={currentVersionData}
+      />
+    );
+
+    const classificationName = screen.getByText('testClassificationDetails');
+
+    expect(classificationName).toBeInTheDocument();
+  });
```

Comment:
Same here we need to check if classification details are rendered by version page or not.

Reply:
Sure